shell/platform/windows/win32_window.cc

```diff
@@ -190,35 +191,71 @@ Win32Window::MessageHandler(HWND hwnd,
         // DefWindowProc will send WM_CHAR for this WM_UNICHAR.
         break;
       }
+      case WM_DEADCHAR:
+      case WM_SYSDEADCHAR:
       case WM_CHAR:
       case WM_SYSCHAR: {
-        if (wparam == VK_BACK)
-          break;
         char32_t code_point = static_cast<char32_t>(wparam);
         static char32_t lead_surrogate = 0;
-        // If code_point is LeadSurrogate, save and return.
+        // If code_point is LeadSurrogate, save to combine to potentially form
+        // a complex Unicode character.
         if ((code_point & 0xFFFFFC00) == 0xD800) {
           lead_surrogate = code_point;
-          return TRUE;
         }
         // Merge TrailSurrogate and LeadSurrogate.
         if (lead_surrogate != 0 && (code_point & 0xFFFFFC00) == 0xDC00) {
           code_point = 0x10000 + ((lead_surrogate & 0x000003FF) << 10) +
                        (code_point & 0x3FF);
         }
         lead_surrogate = 0;
-        window->OnChar(code_point);
+
+        // In an ENG-INTL keyboard, pressing "'" + "e" produces é. In this case,
+        // the "'" key is a dead char, and shouldn't be sent to window->OnChar
+        // for text input. However, the key event should still be sent to
+        // Flutter. The result would be:
+        // * Key event - key code: 222 (quote) - key label: '
+        // * Key event - key code: 69 (e) - key label: é
+        //
+        // As for text input, only the second key press will display a
+        // character.
+        if (wparam != VK_BACK &&
+            (message != WM_DEADCHAR && message != WM_SYSDEADCHAR)) {
+          window->OnChar(code_point);
+        }
+
+        // All key presses that generate a character should be sent from
+        // WM_CHAR. In order to send the full key press information, the keycode
+        // is persisted in a static variable _keycode_for_char_message obtained
+        // from WM_KEYDOWN.
+        const unsigned int scancode = (lparam >> 16) & 0xff;
+        window->OnKey(_keycode_for_char_message, scancode, WM_KEYDOWN,
+                      code_point);
+        _keycode_for_char_message = 0;
```

Comment:
Isn't this only useful if you also wrap this whole block of code in a check that `_keycode_for_char_message` isn't zero?

Reply:
Yeah. `_keycode_for_char_message ` will always be updated since WM_CHAR is always called after WM_KEYDOWN